dlt/sources/rest_api/__init__.py

```diff
@@ -405,14 +410,22 @@ def paginate_dependent_resource(
 
 def _validate_config(config: RESTAPIConfig) -> None:
     c = deepcopy(config)
-    client_config = c.get("client")
+    _mask_client_config_auth(c.get("client"))
+    resources = c.get("resources")
+    if resources:
+        for resource in resources:
+            if isinstance(resource, str) or isinstance(resource, DltResource):
```

Comment:
I would go for the opposite check (is instance of `EndpointResource`?), is there a reason for testing like this

Reply:
Yes, it's a `TypedDict` and the check is not (yet) supported on it :-/

Reply:
sure, you could test only for `dict`, but that should be enough. if's it's a `dict` (even if an incorrect one) the code will try to mask the client field. Then the real validation will happen.

It is not super precise (a comment to mention that the actual test should be for the proper type, but python...), but a bit simpler to read.

Happy to hear other opinions @burnash